acme.go

```diff
@@ -185,6 +185,56 @@ func authorize(client *http.Client, config *Config, domain string) (ChallengeSet
 	return authzresp.ChallengeSet, nil
 }
 
+// Generates a key authorization string for a given token.
+func keyAuthorization(config *Config, token string) string {
+	return fmt.Sprintf("%s.%s", token, jwkThumbprint(config.Key.PublicKey))
+}
+
+// acceptChallenge informs the server that the client accepts one of its challenges for validation.
+// The server will then perform the validation asynchronously.
+//
+// If client argument is nil, DefaultClient will be used.
+func acceptChallenge(client *http.Client, config *Config, challenge Challenge) (Challenge, error) {
+	if client == nil {
+		client = http.DefaultClient
+	}
+	nonce, err := fetchNonce(client, challenge.URI)
+	if err != nil {
+		return Challenge{}, err
+	}
+
+	// prepare challenge request
+	req := struct {
+		Resource         string `json:"resource"`
+		Type             string `json:"type"`
+		KeyAuthorization string `json:"keyAuthorization"`
+	}{
+		Resource:         "challenge",
+		Type:             challenge.Type,
+		KeyAuthorization: keyAuthorization(config, challenge.Token),
```

Comment:
It doesn't seem like this deserves a separate func:

    Auth: challenge.Token + "." + jwkThumbprint(&config.Key.PublicKey),